ipi/interfaces/sockets.py

```diff
@@ -837,6 +844,15 @@ def dispatch_free_client(self, fc, match_ids="any", send_threads=[]):
                 continue
             elif match_ids == "none" and fc.lastreq is not None:
                 continue
+            elif (
+                self.match_mode == "force_match"
+                and match_ids == "none"
+                and (r["id"] in [c.lastreq for c in self.clients])
+            ):
+                # if using forced match mode and the user connects more clients than there are replicas, do not allow this client to
```

Comment:
This could result in not-ideal situations for the user, in which they unknowingly have a client just idling. How is this situation handled?

Reply:
Hi, thanks for the feedback. I'll have a go building a test using one of the driver codes - will let you know in Slack if I run into any difficulties. Regarding the potential idling of excess drivers, I was actually wanting to ask yourselves how you'd like this to be handled. I wondered about forcefully terminating that superfluous driver by passing it an exit signal. However, I was unsure if something more gentle like a simple warning message would be preferable so that the user can handle shutdown of the driver as they wish?